src/services/algorithms_init/AlgorithmsInit_service.h

```diff
@@ -50,12 +50,12 @@ class AlgorithmsInit_service : public JService
             logger.init([this](const algorithms::LogLevel l, std::string_view caller, std::string_view msg){
                 static std::mutex m;
                 std::lock_guard<std::mutex> lock(m);
-                static std::map<std::string_view, std::shared_ptr<spdlog::logger>> loggers;
-                if (! loggers.contains(caller)) {
+                static std::map<std::string, std::shared_ptr<spdlog::logger>> loggers;
+                if (! loggers.contains(std::string(caller))) {
                     this->m_log->debug("Initializing algorithms::LogSvc logger {}", caller);
-                    loggers[caller] = this->m_log_service->logger(std::string(caller));
+                    loggers[std::string(caller)] = this->m_log_service->logger(std::string(caller));
```

Comment:
Somehow this (and below) feels unnecessary: we shouldn't need to create a string from a string_view to do the comparison against the string keys... I think a transparent comparator may work here (but likely still creates a temporary). At definition:
```
                static std::map<std::string, std::shared_ptr<spdlog::logger>, std::less<>> loggers;
```
and here (and below) at use it can remain unchanged:
```suggestion
                    loggers[caller] = this->m_log_service->logger(std::string(caller));
```

Reply:
That wouldn't compile for me.

Reply:
Let's just go with what you have then.